File: shell/src/main/java/com/sequenceiq/cloudbreak/shell/util/TopologyUtil.java
package com.sequenceiq.cloudbreak.shell.util;

import java.util.Set;

import com.sequenceiq.cloudbreak.api.model.TopologyResponse;

public class TopologyUtil {
    private TopologyUtil() {
    }

    public static void checkTopologyForResource(Set<TopologyResponse> publics, Long topologyId, String platform) {
        if (publics != null && topologyId != null) {
            for (TopologyResponse t : publics) {
                if (t.getId().equals(topologyId)) {
                    if (t.getCloudPlatform().equals(platform)) {
                        return;
                    } else {
                        throw new RuntimeException("The selected platform belongs to a different cloudplatform.");
                    }
                }
            }
            throw new RuntimeException("Not found platform with id: " + topologyId);
        }
    }
}

File: shell/src/main/java/com/sequenceiq/cloudbreak/shell/util/TopologyUtil.kt
package com.sequenceiq.cloudbreak.shell.util

import com.sequenceiq.cloudbreak.api.model.TopologyResponse

object TopologyUtil {

    fun checkTopologyForResource(publics: Set<TopologyResponse>?, topologyId: Long?, platform: String) {
        if (publics != null && topologyId != null) {
            var found = false
            for (t in publics) {
                if (t.id == topologyId) {
                    found = true
                    if (t.cloudPlatform == platform) {
                        return
                    } else {
                        throw RuntimeException("The selected platform belongs to a different cloudplatform.")
                    }
                }
            }
            if (!found) {
                throw RuntimeException("Not found platform with id: " + topologyId)
            }
        }
    }
}
